### PR TITLE
[Flow] Optionally fuse the fill in the dequant fusion pass

### PR DESCRIPTION
**Pull request by qedawkins**
Previously fills on the matmul output would not be fused with the rest of the dispatch, leading to separate buffer_fills. Fuse the fills if present.

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FuseDequantizationMatmul.cpp [hunk 1]
```diff
@@ -56,6 +56,20 @@ static LogicalResult fuseDequantAndMatmul(RewriterBase &rewriter,
   if (failed(maybeFusedRegionOp))
     return failure();
 
+  if (fill && *fill) {
+    FailureOr<DispatchRegionOp> maybeFusedFillRegionOp;
+    if (fill.value()->hasOneUse()) {
+      maybeFusedRegionOp =
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FuseDequantizationMatmul.cpp]: Nit: In thoery you dont need to do a separate check for move vs clone. Clone always will basically make the fill dead, but in theorry you shouldnt need to do this anyway.

**Review by MaheshRavishankar — APPROVED** [review]
Should have caught this on the review itself given it is explicitly done so in the lit tests. Thanks

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FuseDequantizationMatmul.cpp (hunk 1 above)
Comment (qedawkins) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FuseDequantizationMatmul.cpp]: I had the same comment before for the dequant and don't really have an opinion here. I'll go with whichever way you prefer.

Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FuseDequantizationMatmul.cpp]: my suggestion is just clone (that has lesser chances of going awry, move has impact on use-def chains). If that works for you, then thats enough. If it doesnt, then add the move.

Comment (qedawkins) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FuseDequantizationMatmul.cpp]: Makes sense. I'll switch the dequant to clone also then.

**Pull request merged**